node/linux/datadog-aas-node-linux/datadog_wrapper

```diff
@@ -1,44 +1,76 @@
-#!/usr/bin/env sh 
+#!/usr/bin/env sh
+
+CURRENT_DIR=$(pwd)
+
+if [ -z "$DD_DIR" ]; then
+    DD_DIR="/home/datadog"
+fi
+
+if [ -z "$DD_BINARIES_URL" ]; then
+    DD_BINARIES_URL="https://gist.github.com/jcstorms1/7e73ab0da121f42d1843dac5b75c21cb/raw"
+fi
+
+setUpCommonEnv() {
+    echo "Creating datadog directory"
+    mkdir -p $DD_DIR && cd $DD_DIR || return
+
+    echo "Downloading required datadog binaries from $DD_BINARIES_URL"
+    curl -L $DD_BINARIES_URL/datadog.tar.gz -o datadog.tar.gz || return
+
+    echo "Decompressing files"
+    tar -xzf datadog.tar.gz || return
+    rm datadog.tar.gz
+
+    echo "Starting trace agent"
+    ./trace-agent &
+    TRACER_PID=$!
+    trap 'kill $TRACER_PID ; exit' INT
+}
 
 setUpNodeEnv() {
     echo "Setting up Datadog tracing for Node"
-    DD_DIR="/home/site/wwwroot/datadog-js"
-    mkdir -p $DD_DIR
 
-    curl -L "https://gist.github.com/jcstorms1/7e73ab0da121f42d1843dac5b75c21cb/raw/datadog-js.tar.gz" -o $DD_DIR/datadog-js.tar.gz
-    tar -xzf $DD_DIR/datadog-js.tar.gz && rm $DD_DIR/datadog-js.tar.gz
-    
-    export NODE_OPTIONS="--require=$DD_DIR/node_modules/dd-trace/init --require=$DD_DIR/start-trace-agent.js"
+    setUpCommonEnv || return
+
+    echo "Installing Node tracer"
+    npm install --prefix $DD_DIR dd-trace || return
+
+    ORIG_NODE_OPTIONS=$NODE_OPTIONS
+    export NODE_OPTIONS="--require=$DD_DIR/node_modules/dd-trace/init $ORIG_NODE_OPTIONS"
+
+    # confirm updates to NODE_OPTIONS
+    node --help >/dev/null || (export NODE_OPTIONS=$ORIG_NODE_OPTIONS && return)
 }
 
 setUpDotnetEnv() {
     echo "Setting up Datadog tracing for .NET"
 
-    echo "Downloading "
-    curl -L "https://gist.github.com/jcstorms1/a712f0df568e81b055b2234b4ec71cd9/raw/datadog-dotnet.tar.gz" -o /home/datadog-dotnet.tar.gz
-    
-    cd /home/
-    tar -xzf datadog-dotnet.tar.gz && rm datadog-dotnet.tar.gz
+    setUpCommonEnv || return
+
+    if [ -z "$DD_DOTNET_TRACER_VERSION" ]; then
+        DD_DOTNET_TRACER_VERSION=2.19.0
+    fi
+    DD_DOTNET_TRACER_FILE=datadog-dotnet-apm-${DD_DOTNET_TRACER_VERSION}.tar.gz
+    DD_DOTNET_TRACER_URL=https://github.com/DataDog/dd-trace-dotnet/releases/download/v${DD_DOTNET_TRACER_VERSION}/${DD_DOTNET_TRACER_FILE}
```

Comment:
The `dd-trace-dotnet` documentation instructs you to download binaries from this same location. https://docs.datadoghq.com/tracing/trace_collection/dd_libraries/dotnet-core?tab=linux#install-the-tracer